src-tauri/src/main.rs

```diff
@@ -34,9 +37,11 @@ async fn get_config(state: tauri::State<'_, MacroBackend>) -> Result<Application
 async fn set_config(
     state: tauri::State<'_, MacroBackend>,
     config: ApplicationConfig,
-) -> Result<(), ()> {
-    state.set_config(config).await;
-    Ok(())
+) -> Result<(), String> {
+    state
```

Comment:
This should propagate the error up to the frontend code

Reply:
Does it need to be an explicit string here? Is it not able to be automatically converted to a String from the anyhow result typing?

Reply:
at least from what I've tried so far, rust does complain about it not being specified and or retyped from anyhow explicitly